tests/unit/test_flask_app.py

```diff
@@ -43,3 +44,48 @@ def test_flask_env(harness: Harness, flask_config: dict):
         f"{FLASK_ENV_CONFIG_PREFIX}{k.upper()}": v if isinstance(v, str) else json.dumps(v)
         for k, v in flask_config.items()
     }
+
+
+@pytest.mark.parametrize(
+    "set_environment_variable, expected",
+    [
+        pytest.param(
+            {"JUJU_CHARM_NO_PROXY": "127.0.0.1"}, {"no_proxy": "127.0.0.1"}, id="no_proxy"
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTP_PROXY": "proxy.test"}, {"http_proxy": "proxy.test"}, id="http_proxy"
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTPS_PROXY": "proxy.test"},
+            {"https_proxy": "proxy.test"},
+            id="https_proxy",
+        ),
+        pytest.param(
+            {"JUJU_CHARM_HTTP_PROXY": "proxy.test", "JUJU_CHARM_HTTPS_PROXY": "proxy.test"},
+            {"http_proxy": "proxy.test", "https_proxy": "proxy.test"},
+            id="http_https_proxy",
+        ),
+    ],
+    indirect=["set_environment_variable"],
+)
```

Comment:
Shouldn't we test the case where we don't set any `JUJU_CHARM_*` proxy variables and we expect no proxy variable in the expected env ?

Reply:
That's right! Added, updated, thanks!